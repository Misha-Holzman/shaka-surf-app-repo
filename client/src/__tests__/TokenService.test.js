
describe('TokenService', () => {
  let TokenService;

  beforeEach(() => {
    TokenService = require('../TokenService');
    localStorage.clear();
  });

  it('should instantiate', () => {
    expect(TokenService).toMatchSnapshot();
  });

  it('should persist a token', () => {
    const myToken = 'my-special-token';

    TokenService.token = myToken;

    expect(TokenService.token).toMatch(myToken);
  });

  it('should NOT persist a previous token', () => {
    expect(TokenService.token).toBeNull();
  });

  it('should destroy a token', () => {
    const myToken = 'my-special-token';

    TokenService.token = myToken;
    TokenService.destroy();

    expect(TokenService.token).toBeNull();
  });

  it('should resolve a string dynamically', () => {
    const myToken = 'my-special-token';

    TokenService.token = myToken;

    const headers = {
      get Authorization() {
        return `Bearer ${TokenService.token}`;
      },
    };

    TokenService.token = 'abc-1234';

    expect(headers.Authorization).toBe('Bearer abc-1234')

  });
});
