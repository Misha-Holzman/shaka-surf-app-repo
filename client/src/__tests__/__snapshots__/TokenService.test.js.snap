// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenService should instantiate 1`] = `
Object {
  "destroy": [Function],
  "token": null,
}
`;
